ViAn/GUI/mainwindow.cpp

```diff
@@ -65,13 +56,38 @@ MainWindow::~MainWindow() {
 
     delete iconOnButtonHandler;
     delete fileHandler;
+    mvideo_player->terminate();
     delete mvideo_player;
     delete ui;
 }
 
 /**
+ * @brief MainWindow::setup_video_player
+ * @param mplayer
+ * Connects all signals and slots that are needed between video_player and mainwindow.
+ */
+void MainWindow::setup_video_player(video_player *mplayer) {
+    QObject::connect(mplayer, SIGNAL(processed_image(QImage)),
+                     this, SLOT(update_video(QImage)));
+    QObject::connect(mplayer, SIGNAL(update_current_frame(int)),
+                     this, SLOT(set_video_slider_pos(int)));
+    QObject::connect(this, SIGNAL(resize_video_frame(int,int)),
+                     mplayer, SLOT(scaling_event(int,int)));
+    QObject::connect(this, SIGNAL(next_video_frame()),
+                     mplayer, SLOT(next_frame()));
+    QObject::connect(this, SIGNAL(prev_video_frame()),
+                     mplayer, SLOT(previous_frame()));
+    QObject::connect(this, SIGNAL(set_play_video()),
+                     mplayer, SLOT(on_play_video()));
+    QObject::connect(this, SIGNAL(set_pause_video()),
+                     mplayer, SLOT(on_pause_video()));
+    QObject::connect(this, SIGNAL(set_stop_video()),
+                     mplayer, SLOT(on_stop_video()));
+}
```

Comment:
Why do you create slots between functions in mainwindow and video_player? Isn't it easier just to call the methods as before, a la mvideo_player->next_frame() for example? Or is this because of thread safety?